File: statsig-php/src/StickyValues.php
<?php

namespace Statsig;

class StickyValues
{
    public bool $value;
    public $json_value;  // Any type, can be null, array, or object
    public string $rule_id;
    public ?string $group_name;
    public array $secondary_exposures;
    public array $undelegated_secondary_exposures;
    public ?string $config_delegate;
    public ?array $explicit_parameters;
    public int $time;
    public ?int $config_version;

    public function __construct(
        bool $value,
        $json_value,
        string $rule_id,
        ?string $group_name,
        array $secondary_exposures,
        array $undelegated_secondary_exposures,
        ?string $config_delegate,
        ?array $explicit_parameters,
        int $time,
        ?int $config_version = null
    ) {
        $this->value = $value;
        $this->json_value = $json_value;
        $this->rule_id = $rule_id;
        $this->group_name = $group_name;
        $this->secondary_exposures = $secondary_exposures;
        $this->undelegated_secondary_exposures = $undelegated_secondary_exposures;
        $this->config_delegate = $config_delegate;
        $this->explicit_parameters = $explicit_parameters;
        $this->time = $time;
        $this->config_version = $config_version;
    }

    public static function fromArray(array $data): self
    {
        return new self(
            $data['value'] ?? false,
            $data['json_value'] ?? null,
            $data['rule_id'] ?? '',
            $data['group_name'] ?? null,
            $data['secondary_exposures'] ?? [],
            $data['undelegated_secondary_exposures'] ?? [],
            $data['config_delegate'] ?? null,
            $data['explicit_parameters'] ?? null,
            $data['time'] ?? 0,
            $data['config_version'] ?? null
        );
    }

    public function toArray(): array
    {
        $result = [
            'value' => $this->value,
            'json_value' => $this->json_value,
            'rule_id' => $this->rule_id,
            'group_name' => $this->group_name,
            'secondary_exposures' => $this->secondary_exposures,
            'undelegated_secondary_exposures' => $this->undelegated_secondary_exposures,
            'config_delegate' => $this->config_delegate,
            'explicit_parameters' => $this->explicit_parameters,
            'time' => $this->time,
        ];

        if ($this->config_version !== null) {
            $result['config_version'] = $this->config_version;
        }

        return $result;
    }
}

File: statsig-java/src/main/java/com/statsig/StickyValues.java
package com.statsig;

import com.alibaba.fastjson2.annotation.JSONCreator;
import com.alibaba.fastjson2.annotation.JSONField;
import java.util.List;
import java.util.Map;

public class StickyValues {
  @JSONField(name = "value")
  public final boolean value;

  @JSONField(name = "json_value")
  public final Map<String, Object> jsonValue;

  @JSONField(name = "rule_id")
  public final String ruleId;

  @JSONField(name = "group_name")
  public final String groupName;

  @JSONField(name = "secondary_exposures")
  public final List<Map<String, String>> secondaryExposures;

  @JSONField(name = "undelegated_secondary_exposures")
  public final List<Map<String, String>> undelegatedSecondaryExposures;

  @JSONField(name = "config_delegate")
  public final String configDelegate;

  @JSONField(name = "explicit_parameters")
  public final List<String> explicitParameters;

  @JSONField(name = "time")
  public final Long time;

  @JSONField(name = "config_version")
  public final Integer configVersion;

  @JSONCreator
  public StickyValues(
      @JSONField(name = "value") boolean value,
      @JSONField(name = "json_value") Map<String, Object> jsonValue,
      @JSONField(name = "rule_id") String ruleId,
      @JSONField(name = "group_name") String groupName,
      @JSONField(name = "secondary_exposures") List<Map<String, String>> secondaryExposures,
      @JSONField(name = "undelegated_secondary_exposures")
          List<Map<String, String>> undelegatedSecondaryExposures,
      @JSONField(name = "config_delegate") String configDelegate,
      @JSONField(name = "explicit_parameters") List<String> explicitParameters,
      @JSONField(name = "time") Long time,
      @JSONField(name = "config_version") Integer configVersion) {
    this.value = value;
    this.jsonValue = jsonValue;
    this.ruleId = ruleId;
    this.groupName = groupName;
    this.secondaryExposures = secondaryExposures;
    this.undelegatedSecondaryExposures = undelegatedSecondaryExposures;
    this.configDelegate = configDelegate;
    this.explicitParameters = explicitParameters;
    this.time = time;
    this.configVersion = configVersion;
  }

  public boolean getValue() {
    return value;
  }

  public Map<String, Object> getJsonValue() {
    return jsonValue;
  }

  public String getRuleId() {
    return ruleId;
  }

  public String getGroupName() {
    return groupName;
  }

  public List<Map<String, String>> getSecondaryExposures() {
    return secondaryExposures;
  }

  public List<Map<String, String>> getUndelegatedSecondaryExposures() {
    return undelegatedSecondaryExposures;
  }

  public String getConfigDelegate() {
    return configDelegate;
  }

  public List<String> getExplicitParameters() {
    return explicitParameters;
  }

  public Long getTime() {
    return time;
  }

  public Integer getConfigVersion() {
    return configVersion;
  }
}
